client/state/account-recovery/reset/reducer.js

```diff
@@ -0,0 +1,38 @@
+/**
+ * External dependencies
+ */
+import { combineReducers } from 'redux';
+
+/**
+ * Internal dependencies
+ */
+import { createReducer } from 'state/utils';
+import {
+	ACCOUNT_RECOVERY_RESET_OPTIONS_ERROR,
+	ACCOUNT_RECOVERY_RESET_OPTIONS_RECEIVE,
+	ACCOUNT_RECOVERY_RESET_OPTIONS_REQUEST,
+	SERIALIZE,
+} from 'state/action-types';
+
+const resetOptions = createReducer( {}, {
+	[ ACCOUNT_RECOVERY_RESET_OPTIONS_REQUEST ]: ( state ) => ( {
+		...state,
+		isRequesting: true,
+	} ),
+	[ ACCOUNT_RECOVERY_RESET_OPTIONS_RECEIVE ]: ( state, { options } ) => ( {
+		...state,
+		isRequesting: false,
+		error: undefined,
+		options,
+	} ),
+	[ ACCOUNT_RECOVERY_RESET_OPTIONS_ERROR ]: ( state, { error } ) => ( {
+		...state,
+		isRequesting: true,
+		error,
+	} ),
+	[ SERIALIZE ]: () => ( {} ),
+} );
+
+export default combineReducers( {
+	options: resetOptions,
+} );
```

Comment:
can I encourage you to think about separating this reducer into multiple reducers each geared towards a specific data item, independently?

right now we have one reducer with a complex initial state and it updates three separate data items, but we have an idiom already for this behavior, one which doesn't require the use of `...state` - `combineReducers()`

what if we had a reducer for `isRequesting`, `error`, and `options`? At that point they would all have a very basic type of data representing their state and would no longer be coupled to each other.

notice how we shift from being action-first to data-first

```js
const isRequesting = ( state = false, action ) => get( {
	[ ACCOUNT_RECOVERY_RESET_OPTIONS_REQUEST ]: true,
	[ ACCOUNT_RECOVERY_RESET_OPTIONS_RECEIVE ]: false,
	[ ACCOUNT_RECOVERY_RESET_OPTIONS_ERROR ]: true,
}, action.type, state );

const error = ( state = null, action ) => get( {
	[ ACCOUNT_RECOVERY_RESET_OPTIONS_RECEIVE ]: null,
	[ ACCOUNT_RECOVERY_RESET_OPTIONS_ERROR ]: action.error,
}, action.type, state );

const options = ( state = {}, action ) =>
	ACCOUNT_RECOVERY_REST_OPTIONS_RECEIVE === action.type
		? action.options
		: state;

const resetOptions = combineReducers( {
	isRequesting,
	error,
	options,
} );
```

We can easily build the initial state here and see the structure of the reducer from the `combineReducers` call. there are a few things in there that are different than the code in this PR. I'd be happy to talk about them if you have questions.